src/main/java/dan200/computercraft/client/gui/widgets/WidgetTerminal.java

```diff
@@ -317,6 +349,22 @@ public void update()
             m_rebootTimer = 0.0f;
             m_shutdownTimer = 0.0f;
         }
+
+        // Handle mouse movement
+        if( m_mouseMoved && System.currentTimeMillis() - ComputerCraft.mouseMoveThrottle > m_lastMouseMove )
+        {
+            IComputer computer = m_computer.getComputer();
+            if( computer != null )
+            {
+                computer.mouseMove(
+                    m_lastMouseX == -1 ? -1 : m_lastMouseX + 1,
+                    m_lastMouseY == -1 ? -1 : m_lastMouseY + 1
+                );
+            }
+
+            m_mouseMoved = false;
+            m_lastMouseMove = System.currentTimeMillis();
```

Comment:
I'd be inclined to put this processing on the server - inside `InputHandler` most likely. The packet is 3 bytes (plus discriminator) so the network traffic is not really my worry - event spam is more problematic.

One can probably drop the delay to something smaller - 50ms maybe? Goodness knows - try it out on SC and see what's tolerable.

Reply:
WRT the -1 event, I'm not sure if you saw my 'Thoughts' up the top, but what do you think about just making it compatible with the other events? So just always send `1` as the first (e.g. button) argument.

Reply:
> Also, does this send the event every frame, or just the first one it moves outside?

It just sends the first time it moves outside, as m_lastMouse[XY] get set to `-1`, and that value is remembered.

Reply:
> One can probably drop the delay to something smaller - 50ms maybe? Goodness knows - try it out on SC and see what's tolerable.

My concern isn't the mouse packets, it's screen redraws. That said, we can drop the default to 50 (0 also works, by the way), and just increase SwitchCraft as needed.

Reply:
> I'd be inclined to put this processing on the server - inside `InputHandler` most likely. The packet is 3 bytes (plus discriminator) so the network traffic is not really my worry - event spam is more problematic.

One small thing - not exactly sure how to fit it in InputHandler because it's an interface and we need a variable